File: client/src/app/_models/Order.ts
import { Client } from './Client';
import { OrderProduct } from './OrderProduct';
import { OrderService } from './OrderService';
import { Status } from './Status';
import { Vehicle } from './Vehicle';

export interface Order {
  id: number;
  orderNumber: string;
  admissionDate?: Date;
  deadlineDate?: Date;
  clientDescription: string;
  fuelLevel: string;
  repairDescription: string;
  invoiceId: string;
  mileage?: number;
  totalJobsNet: number;
  totalJobsGross: number;
  totalPartsNet: number;
  totalPartsGross: number;
  totalNet: number;
  totalGross: number;
  createdAt: Date;
  updatedAt: Date;
  finishDate?: Date;
  statusId: number;
  clientId?: number;
  vehicleId?: number;
  status: Status;
  client: Client;
  vehicle: Vehicle;
  orderServices: Array<OrderService>;
  orderProducts: Array<OrderProduct>;
}

File: API/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{   
    public class Order
    {
        public int Id { get; set; }
        [Required]
        public string OrderNumber { get; set; }
        public DateTime? AdmissionDate { get; set; }
        public DateTime? DeadlineDate { get; set; }
        public string ClientDescription { get; set; }
        public string RepairDescription { get; set; }
        public string InvoiceId { get; set; }
        public int? Mileage { get; set; }
        public float? TotalJobsNet { get; set; }
        public float? TotalJobsGross { get; set; }
        public float? TotalPartsNet { get; set; }
        public float? TotalPartsGross { get; set; }
        public float? TotalNet { get; set; }
        public float? TotalGross { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? FinishDate { get; set; }

        // -------------------- Foreign keys
        public int AppUserId { get; set; }
        public int StatusId { get; set; }
        public int? ClientId { get; set; }      // Change to int cause cascade deletion but I think is better to stay int? 
                                                // because we will be able to check all orders even if corresopnding car or client was deleted
        public int? VehicleId { get; set; }
        public AppUser AppUser { get; set; }
        public Status Status { get; set; }
        public Client Client { get; set; }
        public Vehicle Vehicle { get; set; }
        
    }
}